src/main/java/interfaces/InputValidator.java

```diff
@@ -2,8 +2,17 @@
 
 public interface InputValidator {
     /**
-     * @param input the user input to validate
-     * @return whether the given input is valid
+     * @param input the user input to parse and validate
+     * @return null if the given input is not valid, otherwise the parsed input
      */
-    public boolean validateInput(String input);
+    public String parseAndValidate(String input);
+
+    /**
+     * @param input the invalid user input
+     * @return the corresponding error message
+     */
+    public default String getErrorMessage(String input) {
+        return "This is not one of the available options provided.";
+    }
```

Comment:
In case other InputValidator subclass wants to overwrite this error message method, I figured they might need the actual invalid input to know what specific error message should be generated. Just to keep options open.